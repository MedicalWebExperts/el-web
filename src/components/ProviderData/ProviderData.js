import React from 'react';
import { string } from 'prop-types';
import css from 'styled-jsx/css';
import { faMedkit, faMapMarkerAlt } from '@fortawesome/free-solid-svg-icons';

import Avatar from '../Avatar/Avatar';
import { H3, H4, Text } from '../Typography';
import Icon from '../Icon/Icon';
import theme from '../../styles';

const propTypes = {
  avatar: string.isRequired,
  name: string.isRequired,
  title: string,
  description: string,
  location: string,
  specialty: string,
};

const defaultProps = {
  title: '',
  description: '',
  location: '',
  specialty: '',
};

const style = css`
  div {
    display: flex;
  }
  .innerWrapper {
    flex: 1;
    flex-direction: row;
    align-items: center;
  }
  .infoWrapper {
    flex: 6;
    flex-direction: column;
    padding-left: 20px;
    margin-top: -5px;
  }
  .textWrapper {
    align-items: center;
    padding-bottom: 5px;
  }
  .iconWrapper {
    align-items: baseline;
  }
`;

const customStyles = {
  name: {
    fontSize: '18px',
  },
  comma: {
    fontSize: '18px',
    fontWeight: 400,
    paddingRight: '5px',
  },
  title: {
    fontSize: '18px',
    color: theme.colors.textSecondary,
    fontWeight: 400,
  },
  h4Title: {
    fontSize: '14px',
    color: theme.colors.textSecondary,
    fontWeight: 400,
    paddingRight: 5,
    paddingLeft: 5,
  },
  h4: {
    fontSize: '14px',
    color: theme.colors.textSecondary,
  },
  text: {
    fontSize: 12,
  },
  actionText: {
    fontSize: 14,
    color: theme.colors.primary,
  },
};

const ProviderData = ({
  avatar, name, title, specialty, location, description,
}) => (
  <div className="innerWrapper">
    <style jsx>{style}</style>
    <div>
      <Avatar image={avatar} alt="Provider's image" />
    </div>
    <div className="infoWrapper">
      <div className="textWrapper">
        <H3 text={name} styles={customStyles.name} />
        <H3 text="," styles={customStyles.comma} />
        <H3 text={title} styles={customStyles.title} />
      </div>

      <div className="iconWrapper">
        <Icon
          icon={faMedkit}
          size="sm"
          color={theme.colors.textSecondary}
          height="14px"
          width="14px"
        />
        <H4 text="Specialty:" styles={customStyles.h4Title} />
        <H4 text={specialty} styles={customStyles.h4} />
      </div>
      <div className="iconWrapper">
        <Icon
          icon={faMapMarkerAlt}
          size="sm"
          color={theme.colors.textSecondary}
          height="14px"
          width="14px"
        />
        <H4 text="Location:" styles={customStyles.h4Title} />
        <H4 text={location} styles={customStyles.h4} />
      </div>
      <Text text={description} styles={customStyles.text} />
    </div>
  </div>
);

ProviderData.propTypes = propTypes;
ProviderData.defaultProps = defaultProps;

export default ProviderData;
