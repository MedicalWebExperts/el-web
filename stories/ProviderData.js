import React from 'react';
import { storiesOf } from '@storybook/react';

import { ProviderData } from '../src';

const item = {
  image: 'https://placehold.it/350x350',
  name: 'John Doe',
  title: 'MD',
  specialty: 'Family Medicine',
  location: 'Mansfield Clinic',
  description: 'This is a description, like a subtitle.',
};

export default () => {
  storiesOf('ProviderData', module).add('Default', () => (
    <ProviderData
      avatar={item.image}
      name={item.name}
      title={item.title}
      specialty={item.specialty}
      location={item.location}
      description={item.description}
      secondaryDescription={item.secondaryDescription}
    />
  ));
};
