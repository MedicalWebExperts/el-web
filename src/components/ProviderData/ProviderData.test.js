import React from 'react';
import renderer from 'react-test-renderer';
import ProviderData from './ProviderData';

const defaultProps = {
  image: 'https://placehold.it/350x350',
  name: 'John Doe',
  title: 'MD',
  specialty: 'Family Medicine',
  location: 'Mansfield Clinic',
  description: 'This is a description, like a subtitle.',
};

describe('Component ProviderData Snapshot', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      value: jest.fn(() => ({ matches: false })),
    });
  });

  it('without arguments, take the default values', () => {
    const tree = renderer.create(<ProviderData {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
