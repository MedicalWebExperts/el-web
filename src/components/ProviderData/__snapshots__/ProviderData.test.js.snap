// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProviderData Snapshot without arguments, take the default values 1`] = `
<div
  className="jsx-9121096 innerWrapper"
>
  <div
    className="jsx-9121096"
  >
    <div
      className="jsx-46401105"
    >
      <img
        alt="Provider's image"
        className="jsx-46401105"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="jsx-9121096 infoWrapper"
  >
    <div
      className="jsx-9121096 textWrapper"
    >
      <h3
        className="jsx-958193894"
        style={
          Object {
            "fontSize": "18px",
          }
        }
      >
        John Doe
      </h3>
      <h3
        className="jsx-958193894"
        style={
          Object {
            "fontSize": "18px",
            "fontWeight": 400,
            "paddingRight": "5px",
          }
        }
      >
        ,
      </h3>
      <h3
        className="jsx-958193894"
        style={
          Object {
            "color": "#444444",
            "fontSize": "18px",
            "fontWeight": 400,
          }
        }
      >
        MD
      </h3>
    </div>
    <div
      className="jsx-9121096 iconWrapper"
    >
      <div
        className="jsx-4020043865 icon"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-medkit fa-w-16 fa-sm "
          color="#444444"
          data-icon="medkit"
          data-prefix="fas"
          height="14px"
          role="img"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
          styles={Object {}}
          viewBox="0 0 512 512"
          width="14px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 480h320V128h-32V80c0-26.51-21.49-48-48-48H176c-26.51 0-48 21.49-48 48v48H96v352zm96-384h128v32H192V96zm320 80v256c0 26.51-21.49 48-48 48h-16V128h16c26.51 0 48 21.49 48 48zM64 480H48c-26.51 0-48-21.49-48-48V176c0-26.51 21.49-48 48-48h16v352zm288-208v32c0 8.837-7.163 16-16 16h-48v48c0 8.837-7.163 16-16 16h-32c-8.837 0-16-7.163-16-16v-48h-48c-8.837 0-16-7.163-16-16v-32c0-8.837 7.163-16 16-16h48v-48c0-8.837 7.163-16 16-16h32c8.837 0 16 7.163 16 16v48h48c8.837 0 16 7.163 16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <h4
        className="jsx-610941067"
        style={
          Object {
            "color": "#444444",
            "fontSize": "14px",
            "fontWeight": 400,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Specialty:
      </h4>
      <h4
        className="jsx-610941067"
        style={
          Object {
            "color": "#444444",
            "fontSize": "14px",
          }
        }
      >
        Family Medicine
      </h4>
    </div>
    <div
      className="jsx-9121096 iconWrapper"
    >
      <div
        className="jsx-4020043865 icon"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-map-marker-alt fa-w-12 fa-sm "
          color="#444444"
          data-icon="map-marker-alt"
          data-prefix="fas"
          height="14px"
          role="img"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
          styles={Object {}}
          viewBox="0 0 384 512"
          width="14px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <h4
        className="jsx-610941067"
        style={
          Object {
            "color": "#444444",
            "fontSize": "14px",
            "fontWeight": 400,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Location:
      </h4>
      <h4
        className="jsx-610941067"
        style={
          Object {
            "color": "#444444",
            "fontSize": "14px",
          }
        }
      >
        Mansfield Clinic
      </h4>
    </div>
    <p
      className="jsx-3515229906"
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      This is a description, like a subtitle.
    </p>
  </div>
</div>
`;
